operator/controllers/btpoperator_controller.go

```diff
@@ -502,19 +504,31 @@ func (r *BtpOperatorReconciler) handleDeprovisioning(ctx context.Context) error
 			logger.Info("hard delete success")
 			if err := r.cleanUpAllBtpOperatorResources(ctx, namespaces); err != nil {
 				logger.Error(err, "failed to remove related installed resources")
-				return err
+				return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, ResourceRemovalFailed, "Unable to remove installed resources")
 			}
 		} else {
-			if err := r.handleSoftDelete(ctx, namespaces); err != nil {
+			err := r.UpdateBtpOperatorStatus(ctx, cr, types.StateDeleting, SoftDeleting, "Being soft deleted")
+			if err != nil {
+				logger.Error(err, "failed to update status")
 				return err
 			}
+			if err := r.handleSoftDelete(ctx, namespaces); err != nil {
+				logger.Error(err, "failed to hard delete")
+				return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, HardDeleteFailed, "Hard deleting failed")
+			}
 		}
 	case <-time.After(r.timeout):
 		logger.Info("timeout of hard delete", "duration", r.timeout)
 		timeoutChannel <- true
-		if err := r.handleSoftDelete(ctx, namespaces); err != nil {
+		err := r.UpdateBtpOperatorStatus(ctx, cr, types.StateDeleting, SoftDeleting, "Being soft deleted")
+		if err != nil {
+			logger.Error(err, "failed to update status")
 			return err
 		}
+		if err := r.handleSoftDelete(ctx, namespaces); err != nil {
+			logger.Error(err, "failed to soft delete")
+			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, SoftDeleteFailed, "Soft deleting failed")
```

Comment:
`UpdateBtpOperatorStatus` return values can shadow error returned from `handleSoftDelete`. In such case BtpOperatorReconciler will continue deprovisioning when it shouldn't and clear finalizers from BtpOperator CR

Reply:
The same case in L517

Reply:
Corrected in the next commits. Reason "SoftDeleteFailed" removed